### ISSUE TITLE
On canvas menu: option to hide/show it?

### ISSUE THREAD

**Issue by MartinDM** [FrDH/jQuery.mmenu#723]
Is there a way to toggle this take-over style menu on and off like the off-canvas? It's not included in any of the examples for some reason; it's just instantly initialized and takes over the view...

**Comment by FrDH** [comment]
I'm not sure I understand the question, but if you want an on-canvas menu, there is no opening or closing the menu. The menu will simply fill up the available width and height of its parent HTML node.
You could ofcourse manually use jQuery to show/hide the menu by adding some class that'll set `display: block;` or something like that:
````js
$("#my-button").click(function() {
    $("#my-menu").addClass( "show-the-menu" );
});
````

**Comment by MartinDM** [comment]
OK. Yes I'll do it this way.
It just seemed logical that that even a full-screen menu should be toggle-able in my opinion, even if it means the menu drops down from its position in the DOM (eg if you wanted a bar along the top with the toggle button in it).

Thank you though.

**Comment by FrDH** [comment]
If you do want off-canvas behavior, you could also try to fiddle around with the [off-canvas configuration options](http://mmenu.frebsite.nl/documentation/core/off-canvas.html) to create the desired behavior:
````js
$("#my-menu").mmenu({
    // options
}, {
   // config
   offCanvas: {
      menuInsertMethod: "appendTo",
      menuInsertSelector: "#my-menu-wrapper-div",
      pageSelector: "#my-page-div"
   }
});
````
This will:
1) append the menu to the HTML node with `id="my-menu-wrapper-div"`.
2) set the HTML node with `id="my-page-div"` as the "page".

**Comment by MartinDM** [comment]
OK thanks, will try this approach.
Can I ask also, is the off-canvas option the 'default'? Just noticing that the 'offcanvas' has a lot of SCSS files that need including, off-canvas does not? Should I be including all of those?

**Comment by FrDH** [comment]
I'll try to explain how the plugin is set up:

The plugin is a combination of 
1) the "core" (on-canvas behavior)
2) "core" add-ons (off-canvas, screenreader and scroll-bug-fix).
This is what's called [the core](http://mmenu.frebsite.nl/documentation/core/).

It can be extended with [add-ons](http://mmenu.frebsite.nl/documentation/addons/) and [extensions](http://mmenu.frebsite.nl/documentation/extensions/).

So basically, the off-canvas add-on is included in the core and it's options are set to be used by default. To not use the off-canvas add-on, you can just set the option to `false`.

Hope this helps.

**Comment by MartinDM** [comment]
Hi Fred

Another thing I noticed; when implementing the hamburger menu as on the tutorial:
http://mmenu.frebsite.nl/tutorials/animated-hamburger.html the event causes an error on the console and doesn't fire the plugin, presumably because the button is not wrapped in an a tag as on the mmenu site.
Is it necessary for the toggle link to have a href at all, as removing it produces an error in the library itself:
`Cannot read property 'top' of undefined`

![image](https://cloud.githubusercontent.com/assets/7467069/24859478/a0088790-1de8-11e7-8cfc-e3916f50c789.png)

Bindings look like this 


` $toggleBtn.on("click", function () {
        API.open();
        console.log(API)
    });

    API.bind("opened", function () {
        console.log('opened')
        setTimeout(function () {
            $toggleBtn.addClass("is-active");
        }, 100);
    });
    API.bind("closed", function () {
        console.log('closed')
        setTimeout(function () {
            $toggleBtn.removeClass("is-active");
        }, 100);
    });`

Console logs fine, fails after that.
Does the toggle link have to have a href in order to work? The `is-active` doesn't get added to the toggle button presumably because of the blocking error. Am I doing something wrong following this please?

Thanks in advance.
